### PR TITLE
fix(broker-core): ignore empty or invalid task headers

### LINKED ISSUE #1902: Deploying BPMN file with empty header kills broker
If you have this in a bpmn file and try to deploy it to the broker:

```
          <zeebe:header />
```

You get this error on the broker:

```
05:04:55.477 [zb-stream-processor] [172.30.0.3:26501-zb-actors-0] ERROR io.zeebe.logstreams - Stream processor 'zb-stream-processor' failed to process event. It stop processing further events.
java.lang.NullPointerException: null
	at io.zeebe.util.StringUtil.getBytes(StringUtil.java:31) ~[zb-util-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.util.StringUtil.getBytes(StringUtil.java:27) ~[zb-util-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.util.buffer.BufferUtil.wrapString(BufferUtil.java:51) ~[zb-util-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.workflow.model.transformation.transformer.ServiceTaskTransformer.lambda$encode$0(ServiceTaskTransformer.java:101) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:1.8.0_131]
	at io.zeebe.broker.workflow.model.transformation.transformer.ServiceTaskTransformer.encode(ServiceTaskTransformer.java:99) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.workflow.model.transformation.transformer.ServiceTaskTransformer.transformTaskHeaders(ServiceTaskTransformer.java:83) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.workflow.model.transformation.transformer.ServiceTaskTransformer.transform(ServiceTaskTransformer.java:59) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.workflow.model.transformation.transformer.ServiceTaskTransformer.transform(ServiceTaskTransformer.java:39) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.workflow.model.transformation.TransformationVisitor.visit(TransformationVisitor.java:51) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.model.bpmn.traversal.TypeHierarchyVisitor.visit(TypeHierarchyVisitor.java:39) ~[zb-bpmn-model-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.model.bpmn.traversal.ModelWalker.walk(ModelWalker.java:60) ~[zb-bpmn-model-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.workflow.model.transformation.BpmnTransformer.transformDefinitions(BpmnTransformer.java:99) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.workflow.state.WorkflowPersistenceCache.updateInMemoryState(WorkflowPersistenceCache.java:129) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.workflow.state.WorkflowPersistenceCache.lookupPersistenceStateForWorkflowByKey(WorkflowPersistenceCache.java:246) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.workflow.state.WorkflowPersistenceCache.getWorkflowByKey(WorkflowPersistenceCache.java:237) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.workflow.state.WorkflowState.getWorkflowByKey(WorkflowState.java:54) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.workflow.processor.deployment.TransformingDeploymentCreateProcessor.createTimerIfTimerStartEvent(TransformingDeploymentCreateProcessor.java:94) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.workflow.processor.deployment.TransformingDeploymentCreateProcessor.processRecord(TransformingDeploymentCreateProcessor.java:71) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.logstreams.processor.TypedRecordProcessor.processRecord(TypedRecordProcessor.java:56) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.broker.logstreams.processor.TypedStreamProcessor$DelegatingEventProcessor.processEvent(TypedStreamProcessor.java:170) ~[zeebe-broker-core-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.logstreams.processor.StreamProcessorController.processEvent(StreamProcessorController.java:304) ~[zb-logstreams-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.logstreams.processor.StreamProcessorController.readNextEvent(StreamProcessorController.java:287) ~[zb-logstreams-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:90) [zb-util-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:53) [zb-util-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:187) [zb-util-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:154) [zb-util-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:135) [zb-util-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.run(ActorThread.java:112) [zb-util-0.15.0-SNAPSHOT.jar:0.15.0-SNAPSHOT]
```

And the broker dies completely.

### PR DESCRIPTION
**Pull request by miguelpires**
Changes:
* Ignore empty or invalid task headers while logging it in the broker

closes #1902

### CODE REVIEW

File: broker-core/src/main/java/io/zeebe/broker/workflow/model/transformation/transformer/ServiceTaskTransformer.java [hunk 1]
```diff
@@ -80,34 +82,57 @@ private void transformTaskHeaders(ServiceTask element, final ExecutableServiceTa
...
-      serviceTask.setEncodedHeaders(encodedHeaders);
+      final List<ZeebeHeader> validHeaders =
+          taskHeaders
+              .getHeaders()
+              .stream()
+              .filter(this::isValidHeader)
+              .collect(Collectors.toList());
+
+      if (validHeaders.size() < taskHeaders.getHeaders().size()) {
+        LOG.warn("Found task headers with no key or value. Ignoring");
```
Comment (npepinpe) [inline broker-core/src/main/java/io/zeebe/broker/workflow/model/transformation/transformer/ServiceTaskTransformer.java]: We don't care which headers where invalid?

File: broker-core/src/main/java/io/zeebe/broker/workflow/model/transformation/transformer/ServiceTaskTransformer.java [hunk 2]
```diff
@@ -80,34 +82,57 @@ private void transformTaskHeaders(ServiceTask element, final ExecutableServiceTa
...
+      }
+
+      if (validHeaders != null && !validHeaders.isEmpty()) {
```
Comment (npepinpe) [inline broker-core/src/main/java/io/zeebe/broker/workflow/model/transformation/transformer/ServiceTaskTransformer.java]: In which case would `validHeaders` be null? Even if we filter an empty list, we still get an empty list, no?

File: broker-core/src/test/java/io/zeebe/broker/workflow/activity/ActivityTest.java [hunk 3]
```diff
@@ -167,6 +168,35 @@ public void shouldUnsubscribeFromBoundaryEventTriggersOnTerminating() {
         WorkflowInstanceIntent.ELEMENT_TERMINATING, WorkflowInstanceIntent.ELEMENT_TERMINATED);
   }
 
+  @Test
+  public void shouldIgnoreTaskHeadersIfEmptyOrNull() {
```
Comment (npepinpe) [inline broker-core/src/test/java/io/zeebe/broker/workflow/activity/ActivityTest.java]: Can we split this into two tests, one testing empty, and the other testing null? Ideally we'd test also when the key and values are valid (i.e. key is empty or null, but value isn't, and vice versa).

**Review by npepinpe — CHANGES_REQUESTED** [review]
Looks good, only 2 things.

**Comment by MiguelPires** [comment]
Thanks for the feedback. I updated the code to fix those issues

**Review by npepinpe — APPROVED** [review]
Looks good :+1:

**Pull request merged**